src/session.cpp

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include <dbus.hpp>
+#include <phosphor-logging/elog-errors.hpp>
+#include <phosphor-logging/log.hpp>
+#include <session.hpp>
+
+#include <iostream>
+
+namespace obmc
+{
+namespace session
+{
+
+using namespace phosphor::logging;
+
+void SessionItem::setSessionMetadata(std::string username,
+                                     std::string remoteIPAddr)
+{
+    this->adjustSessionOwner(username);
+    if (remoteIPAddr.empty())
+    {
+        throw InvalidArgument();
+    }
+    this->remoteIPAddr(remoteIPAddr);
+}
+
+void SessionItem::adjustSessionOwner(const std::string& userName)
+{
+    using DBusGetObjectOut = std::map<std::string, std::vector<std::string>>;
+
+    constexpr const std::array userObjectIfaces = {
+        "xyz.openbmc_project.User.Attributes"};
+    auto userObjectPath = "/xyz/openbmc_project/user/" + userName;
+
+    DBusGetObjectOut getUserObject;
+    auto callMethod =
+        bus.new_method_call("xyz.openbmc_project.ObjectMapper",
+                            "/xyz/openbmc_project/object_mapper",
+                            "xyz.openbmc_project.ObjectMapper", "GetObject");
+    callMethod.append(userObjectPath.c_str(), userObjectIfaces);
+    bus.call(callMethod).read(getUserObject);
```

Comment:
I guess this throws an exception if the user doesn't exist.
Are you sure that we shouldn't catch it here?

Reply:
I guess if a user is not found, the obmc object-mapper returns 0 entries, and that doesn't throw an exception.

Reply:
I know that `ObjectMapper` throws an exception in this case.
And I guess that `session-manager` will crash.
It is an unlikely but still possible case.
For example, if `admin` drops the user while he is logging in.

Reply:
I see now that you catch this exception in `SessionManager::create`.
So, let it be as is.